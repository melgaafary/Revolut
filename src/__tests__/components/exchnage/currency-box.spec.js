import React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import CurrencyBox from '../../../components/exchange/currency-box';

const availablePockets = {
  USD: {
    id: 'USD',
    balance: 100,
  },
  GBP: {
    id: 'GBP',
    balance: 210,
  },
  CAD: {
    id: 'CAD',
    balance: 2,
  },
  AUD: {
    id: 'AUD',
    balance: 200,
  },
};
const pocket = availablePockets.USD;

describe('[Component] CurrencyBox', () => {
  const mockOnChange = jest.fn();
  it('should match the snapshot', () => {
    const component = (
      <CurrencyBox
        onCurrencyChange={mockOnChange}
        pocket={pocket}
        availablePockets={availablePockets}
      />
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should show 4 currencies when currency selection button is clicked', () => {
    const component = mount(
      <CurrencyBox
        onCurrencyChange={mockOnChange}
        pocket={pocket}
        availablePockets={availablePockets}
      />,
    );
    component.find('button').simulate('click');
    expect(component.find('button div span')).toHaveLength(4);
    expect(
      component
        .find('button div span')
        .first()
        .text(),
    ).toEqual('USD');
    expect(
      component
        .find('button div span')
        .last()
        .text(),
    ).toEqual('AUD');
  });
  it('should display the correct currency and balance', () => {
    const component = mount(
      <CurrencyBox
        onCurrencyChange={mockOnChange}
        pocket={pocket}
        availablePockets={availablePockets}
      />,
    );
    expect(
      component
        .find('span')
        .first()
        .text(),
    ).toEqual('USD');
    expect(
      component
        .find('span')
        .last()
        .text(),
    ).toEqual('Balance: 100.00 $');
  });
});
