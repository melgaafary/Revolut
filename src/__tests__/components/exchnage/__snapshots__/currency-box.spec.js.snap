// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] CurrencyBox should match the snapshot 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 dxtVvB"
>
  <div
    className="StyledBox-sc-13pk1d4-0 hErwNa"
  >
    <span
      className="StyledText-sc-1sadyjn-0 jpqZaM"
      size="xlarge"
    >
      USD
    </span>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffjnxm"
    />
    <button
      className="StyledButton-sc-323bzc-0 dnCEuz"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <svg
        aria-label="Down"
        className="StyledIcon-ofa7kd-0 jXwebn"
        viewBox="0 0 24 24"
      >
        <polyline
          fill="none"
          points="7.086 3.174 17.086 13.174 7.086 23.174"
          stroke="#000"
          strokeWidth="2"
          transform="scale(1 -1) rotate(-89 -1.32 0)"
        />
      </svg>
    </button>
  </div>
  <span
    className="StyledText-sc-1sadyjn-0 ejzSet"
    size="large"
  >
    Balance: 
    100.00
     
    $
  </span>
</div>
`;
